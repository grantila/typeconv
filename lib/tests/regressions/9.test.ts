import { getJsonSchemaReader, getOpenApiWriter, makeConverter } from "../.."

const testObject = {
	"type": "object",
	"properties": {
		"name": { "minLength": 5, "maxLength": 10, "type": "string" },
		"creationDate": { "format": "date-time", "type": "string" }
	},
	"required": ["name", "creationDate"]
}

const schemas = {
	"definitions": {
		"testType": testObject
	}
};

describe( 'regressions', ( ) =>
{
	it( 'issue #9', async ( ) =>
	{
		const reader = getJsonSchemaReader( );
		const writer = getOpenApiWriter( {
			format: 'yaml',
			title: 'Testconverter',
			version: 'v1',
			schemaVersion: '3.0.0',
		} );
		const { convert } = makeConverter( reader, writer );
		const { data: output } =
			await convert( { data: JSON.stringify( schemas ) } );

		expect( output ).toMatchSnapshot( );
	} );
} );
