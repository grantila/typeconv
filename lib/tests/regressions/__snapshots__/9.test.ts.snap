// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regressions issue #9 1`] = `
"openapi: 3.0.0
info:
  title: Testconverter
  version: v1
paths: {}
components:
  schemas:
    testType:
      properties:
        name:
          minLength: 5
          maxLength: 10
          type: string
        creationDate:
          format: date-time
          type: string
      required:
        - name
        - creationDate
      type: object
"
`;
